packages/paste-website/src/components/customization-landing-page/image-slider/ImageSlider.tsx

```diff
@@ -0,0 +1,185 @@
+import {convertPositionToInputValue, clampValueToRange} from './utils';
+import {useSvgResize} from './useSvgResize';
+
+const INPUT_ID = 'input-range-id';
+const CLIP_PATH_ID = 'clip-path-id';
+
+export const ImageSlider: React.FC<{frontFluidObject: FluidObject; backFluidObject: FluidObject}> = ({
+  frontFluidObject,
+  backFluidObject,
+}) => {
+  const uidSeed = useUIDSeed();
+
+  // Set the initial state of the slider to be roughly at the 60% position.
+  const [value, setValue] = React.useState<number>(MAX_VALUE * 0.6);
+  const [shouldMeasure, setShouldMeasure] = React.useState(false);
+  const [refsAreInitiated, setRefsAreInitiated] = React.useState(false);
+
+  const containerRef = React.useRef<HTMLElement>();
+  const svgCircleRef = React.useRef<SVGCircleElement>();
+
+  const handleMouseUp = (): void => {
+    setShouldMeasure(false);
+  };
+
+  const {width: containerWidth, x: containerX, height: containerHeight} = containerRef?.current
+    ? containerRef.current.getBoundingClientRect()
+    : {width: undefined, height: undefined, x: undefined};
+
+  const {svgOffset, svgHeight, svgWidth} = useSvgResize(containerHeight, containerWidth);
+
+  // Minimum change defines the absolute distance threshold for value to change.
+  // Calculated by half the width of the SVG, default is half of the absolute width.
+  const minimumChange = svgWidth ? svgWidth / 2 : DEFAULT_MIN_CHANGE;
+
+  // Width of SVG Clip used to conditionally show/hide part of the each image
+  // SVG clip is calculated from right to left, since the dynamic value here is the width.
+  // const clip = (value / MAX_VALUE) * (containerWidth as number);
+  const clip = containerWidth && (value / MAX_VALUE) * (containerWidth as number);
+
+  React.useEffect(() => {
+    if (!refsAreInitiated) {
+      setRefsAreInitiated(true);
+    }
+  }, [refsAreInitiated]);
+
+  React.useEffect(() => {
+    return () => {
+      document.removeEventListener('mouseup', handleMouseUp);
+    };
+  }, []);
+
+  return (
+    <Box
+      display={['none', 'block']}
+      minHeight={['size0', '260px', '460px']}
+      maxWidth="size60"
+      position="absolute"
+      top="space150"
+      ref={containerRef as MutableRefObject<HTMLElement>}
+      right="spaceNegative150"
+      width="60%"
+      zIndex="zIndex10"
+      marginRight={['space0', 'space0', 'space100']}
+    >
+      <Box
+        as="label"
+        position="absolute"
+        width="100%"
+        height="100%"
+        display="inherit"
+        maxWidth="640px"
+        cursor={shouldMeasure ? 'grabbing' : 'grab'}
+        aria-label="Controls the width of the customizable image."
+        // @ts-expect-error "htmlFor" is an allowed attribute for Label elements.
+        htmlFor={uidSeed(INPUT_ID)}
+        color="colorTextWeaker"
+        _hover={{color: 'colorTextWeak'}}
+        onMouseDown={() => {
+          setShouldMeasure(true);
+          document.addEventListener('mouseup', handleMouseUp);
```

Comment:
Ah so you are saying calling `removeEventListener` when `isDragging` changes from `true` to `false` is not necessarily 1:1 because of batched state updates? So is it "safer" to remove here to ensure proper clean up?

Also curious why the `removeEventListener` call in the `useEffect` unmount logic was removed. I added that thinking that before the component unmounts we should do cleanup to be safe. Is that risky or just unncessary?

Reply:
> Ah so you are saying calling removeEventListener when isDragging changes from true to false is not necessarily 1:1 because of batched state updates? So is it "safer" to remove here to ensure proper clean up?

Not quite. Ignore `isDragging` for the time being. Previously this code called `addEventListener` more frequently than `removeEventListener`.  `removeEventListener` was only being called in a `useEffect` (so on unmount, aka once), and `addEventListener` was called 0..n times every time someone `mouseDown`

> Also curious why the removeEventListener call in the useEffect unmount logic was removed. I added that thinking that before the component unmounts we should do cleanup to be safe. Is that risky or just unncessary?

Since we're now binding to the document.onMouseUp event, we don't need it.